src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -118,6 +118,35 @@ class ApplicationEntity(EntityBase[ApplicationEntityModel]):
     A Native App application object, created from an application package.
     """
 
+    @property
+    def project_root(self):
+        return self._workspace_ctx.project_root
+
+    @property
+    def package_entity_id(self):
+        return self._entity_model.from_.target
+
+    @property
+    def name(self):
+        return self._entity_model.fqn.name
+
+    @property
+    def role(self):
+        model = self._entity_model
+        return (model.meta and model.meta.role) or self._workspace_ctx.default_role
+
+    @property
+    def warehouse(self):
+        model = self._entity_model
+        return (
+            model.meta and model.meta.warehouse and to_identifier(model.meta.warehouse)
+        ) or to_identifier(self._workspace_ctx.default_warehouse)
+
+    @property
+    def post_deploy_hooks(self):
```

Comment:
looks like this can return a None value. Should it always return a valid list, even if empty, instead?

Reply:
Does it make sense to return Optional[List[...]] here then or did we skip annotating this on purpose?

Reply:
oops, updated